### PR TITLE
feat: user-profile

### LINKED ISSUE #14: Add `handle` field to `users`
I'm not sure if it should go on the actual users table, or if we should create a tertiary `user_informations` table?

I'd also consider naming it something other than `username`, just cause our primary identification method will remain `email` and username kinda implies otherwise. Perhaps `handle`?

Should probably be not null and unique, blah blah blah

### PR DESCRIPTION
**Pull request by benmelz**
Closes #14
Based off #15 (review that first)

### CODE REVIEW

**Comment by hugoamaralsantana** [comment]
I approved but it looks like there may be a few commit overlaps between this and the PSQL ticket. (unless Github is just bugging due to the PSQL ticket not being pushed yet)

**Comment by benmelz** [comment]
Indeed, I'm re-organizing my PR house of cards right now

File: app/models/user.rb [hunk 1]
```diff
@@ -2,4 +2,10 @@
 
 class User < ApplicationRecord
   devise :database_authenticatable, :validatable
+
+  has_one :profile, dependent: :destroy, required: true, validate: true
```
Comment (hugoamaralsantana) [inline app/models/user.rb:6]: what's up with the dependent and destroy field?

File: factories/user/profiles.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
```
Comment (hugoamaralsantana) [inline factories/user/profiles.rb:3]: tldr on what this does?

File: app/models/user.rb (hunk 1 above)
Comment (benmelz) [reply app/models/user.rb:6]: The dependent option defines behavior of "what should happen to the associated record(s) when this one is destroyed?" ("associated" being the profile, and this being the "user").

The options are generally either
- `:nullify` - leave the other one, but set its `_id` field to `nil`
- `:destroy` - delete the other one too
- `:restrict_with_error` - don't allow this one to be destroyed and add an error to the model

File: factories/user/profiles.rb (hunk 2 above)
Comment (benmelz) [reply factories/user/profiles.rb:3]: It's a part of `FactoryBot` which is a framework to assist with generating fake data. It's used by the development seeds, as well as in tests.

The point is basically "autogenerate attributes implicitly so we don't have to do stuff like `User.create email: 'we-have-to', password: 'define-these-every-time'`". Instead you just `FactoryBot.create(:user, ... and only specify the stuff you need to constrain for your test/seed)`

File: app/models/user.rb (hunk 1 above)
Comment (hugoamaralsantana) [reply app/models/user.rb:6]: Ah interesting. That's nice for maintaining the data set uniform without any extra work

File: factories/user/profiles.rb (hunk 2 above)
Comment (hugoamaralsantana) [reply factories/user/profiles.rb:3]: Wow, I wish I had that for my company's tests

**Review by hugoamaralsantana — APPROVED** [review]

**Pull request merged**